matchmake-extension/auto_matchmake_postpone.go

```diff
@@ -0,0 +1,102 @@
+package matchmake_extension
+
+import (
+	"math"
+
+	"github.com/PretendoNetwork/mario-kart-7-secure/globals"
+	nex "github.com/PretendoNetwork/nex-go"
+	match_making "github.com/PretendoNetwork/nex-protocols-go/match-making"
+	matchmake_extension "github.com/PretendoNetwork/nex-protocols-go/matchmake-extension"
+	"github.com/PretendoNetwork/nex-protocols-go/notifications"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/globals"
+	"encoding/json"
+)
+
+func AutoMatchmake_Postpone(err error, client *nex.Client, callID uint32, matchmakeSession *match_making.MatchmakeSession, message string) {
+	missingHandler := false
+	if commonMatchmakeExtensionProtocol.CleanupSearchMatchmakeSessionHandler == nil {
+		logger.Warning("MatchmakeExtension::AutoMatchmake_Postpone missing CleanupSearchMatchmakeSessionHandler!")
+		missingHandler = true
+	}
+	if missingHandler {
+		return
+	}
+	//This is the best way I found to copy the full object. And I still hate it.
+	tmp, _ := json.Marshal(matchmakeSession)
+	matchmakeSessionCopy := match_making.NewMatchmakeSession()
+	json.Unmarshal(tmp, &matchmakeSessionCopy)
```

Comment:
Is this what you asked about on Discord for earlier? I think the best way forward in that case is with the example I gave earlier, by implementing a `Copy` method to `StructureInterface` and implementing that on each structure. We don't have to do this all at once, we can add this method incrementally as we have types which need to be copied

Because I agree, I do not like this at all, and this is likely much slower than just creating a new struct and moving data